server/controllers/api/videos/index.ts

```diff
@@ -184,17 +200,46 @@ async function addVideo (req: express.Request, res: express.Response) {
 
   const videoPhysicalFile = req.files['videofile'][0]
   const videoInfo: VideoCreate = req.body
+  const files = req.files
+
+  return addVideo(req, res, { videoPhysicalFile, videoInfo, files })
+}
+
+async function addVideoResumable (req: express.Request, res: express.Response) {
+  interface VideoPhysicalFile {
+    duration: number
+    filename: string
+    size: number
+    path: string
+    metadata: VideoCreate & express.FileUploadMetadata
+  }
+
+  const videoPhysicalFile = res.locals.videoFileResumable as VideoPhysicalFile
+  const videoInfo = videoPhysicalFile.metadata
+  const files = { bg: { path: await getTmpPath(videoPhysicalFile.filename) } }
+
+  return addVideo(req, res, { videoPhysicalFile, videoInfo, files })
+}
+
+async function addVideo (req: express.Request, res: express.Response, parameters: {
+  videoPhysicalFile: { duration: number, filename: string, size: number, path: string }
+  videoInfo: VideoCreate | VideoCreate & express.FileUploadMetadata
+  files
+}) {
+  const { videoPhysicalFile, videoInfo, files } = parameters
+  const videoChannel = res.locals.videoChannel
+  const user = res.locals.oauth.token.User
 
-  const videoData = buildLocalVideoFromReq(videoInfo, res.locals.videoChannel.id)
+  const videoData = buildLocalVideoFromReq(videoInfo as VideoCreate, videoChannel.id)
   videoData.state = CONFIG.TRANSCODING.ENABLED ? VideoState.TO_TRANSCODE : VideoState.PUBLISHED
   videoData.duration = videoPhysicalFile['duration'] // duration was added by a previous middleware
 
   const video = new VideoModel(videoData) as MVideoFullLight
-  video.VideoChannel = res.locals.videoChannel
+  video.VideoChannel = videoChannel
   video.url = getLocalVideoActivityPubUrl(video) // We use the UUID, so set the URL after building the object
 
   const videoFile = new VideoFileModel({
-    extname: extname(videoPhysicalFile.filename),
+    extname: videoPhysicalFile.filename ? extname(videoPhysicalFile.filename) : extname(videoInfo.name),
```

Comment:
You don't have `.filename` for a resumable upload? If no, could we add a `extension` field to the resumable upload metadata? Because I'm not confident in guessing the extension using `videoInfo.name`

Reply:
Me neither! That's the main cause of errors, as we use names without extensions in the tests (except for `resumable` upload ones, ofc).

Reply:
I managed to fix it by properly passing the `filename` body parameter, which was properly injected by `ngx-upload` but was missing from my test helper.